### ISSUE TITLE
add damping factor scheduler for LM and other optmizers.

### ISSUE THREAD

**Issue by wang-chen** [pypose/pypose#75]
### Problem to solve
Different from 1st order optimizers like [`torch.SGD`](https://pytorch.org/docs/stable/generated/torch.optim.SGD.html?highlight=sgd#torch.optim.SGD), 2nd order optimizers like [`pypose.LM`](https://pypose.org/docs/preview/pr/69/generated/pypose.optim.LM/) require a damping factor instead of a learning rate (`lr`), hence a damping factor scheduler will be required.

### Further details
We would like to first provide a simple damping factor scheduler like what is described in Wikipedia, section [`Choice of damping parameter`](https://en.wikipedia.org/wiki/Levenberg%E2%80%93Marquardt_algorithm). For example:

An effective strategy for the control of the damping parameter, called *delayed gratification*, consists of increasing the parameter by a small amount for each uphill step, and decreasing by a large amount for each downhill step. The idea behind this strategy is to avoid moving downhill too fast in the beginning of optimization, therefore restricting the steps available in future iterations and therefore slowing down convergence. An increase by a factor of 2 and a decrease by a factor of 3 has been shown to be effective in most cases, while for large problems more extreme values can work better, with an increase by a factor of 1.5 and a decrease by a factor of 5.

We would like to set the uphill and downhill factors as a hyperparameter for the damping factor scheduler.

### Proposal
- A simple implementation is to subclass [`torch.optim. _LRScheduler`](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html) and provide a scheduler like [`torch.optim.ReduceLROnPlateau`](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html). However, it uses the interface of `lr` not `damping`. To resue its codes and change the damping factor in each epoch, we need to redefine the internal class method `get_lr `, which will make the codes confusing, as we use it to change the damping factor.

- Another way is to redefine a general scheduler called `pypose.optim._GeneralScheduler` following the definition of `_LRScheduler`, which will also allow the PyPose users to redefine their own schedulers.
